### DISCUSSION TITLE
Toolbar button order

### DISCUSSION THREAD

**Discussion by Flo2410** [igordanchenko/yet-another-react-lightbox#102] [Q&A] [answered]
Hi, 
first of all, thank you for that amazing component!

I would like to know if there is any way to customize the order of the buttons in the toolbar? 

Here my situation: 
I am using the fullscreen plugin and want to add a custom share button to the left of it. 

Currently, it looks like that:
![Screenshot_20230331_140451](https://user-images.githubusercontent.com/31217036/229115476-24fbbab9-99d6-4b65-823b-1c21b1926ef9.png)

with the following toolbar props:
```jsx
<Lightbox
toolbar={{
  buttons: [
    <button type="button" className="yarl__button" key={"share_button"}>
      <BiShareAlt className="p-1 yarl__icon" />
    </button>,
    "close",
  ],
}}
plugins={[Fullscreen, Captions]}
// more config
/>
```
Unfortunately, I couldn't find a way to display the share button left of the fullscreen one.

I would like it to look like this:
![Screenshot_20230331_141047](https://user-images.githubusercontent.com/31217036/229116524-ddff2133-6f09-4328-a4c7-05523c2122e3.png)

Is there any way to do this?

**Comment by igordanchenko** [comment, marked as answer]
Thank you for posting this great question. As of today, there is no elegant way to change the order of buttons added through the `toolbar.buttons` prop and the ones added by plugins. Each plugin prepends its button to the beginning of the `toolbar.buttons` array, so you can easily re-arrange plugin buttons by changing the order of the plugins. However, plugin buttons would always appear to the left of the buttons you add through the `toolbar.buttons` prop, and the only way to change this is to add your own buttons via a custom plugin as well. Here is how you'd do it.

```tsx
import * as React from "react";
import { Lightbox, Plugin } from "yet-another-react-lightbox";
import { useLightboxState } from "yet-another-react-lightbox/core";
import Captions from "yet-another-react-lightbox/plugins/captions";
import Fullscreen from "yet-another-react-lightbox/plugins/fullscreen";
import { BiShareAlt } from "react-icons/bi";

function ShareButton({ onClick }: { onClick: (currentIndex: number) => void }) {
  const { slides, currentIndex } = useLightboxState().state;

  return (
    <button
      type="button"
      className="yarl__button"
      disabled={slides.length === 0}
      onClick={() => onClick(currentIndex)}
    >
      <BiShareAlt className="p-1 yarl__icon" />
    </button>
  );
}

export default function Page() {
  const [open, setOpen] = React.useState(false);

  const onClick = (currentIndex: number) => {
    // handle Share button click
  };

  const Share: Plugin = ({ augment }) => {
    augment(({ toolbar: { buttons, ...restToolbar }, ...rest }) => ({
      toolbar: { buttons: [<ShareButton key="share" onClick={onClick} />, ...buttons], ...restToolbar },
      ...rest,
    }));
  };

  return (
    <>
      <Lightbox
        open={open}
        close={() => setOpen(false)}
        slides={slides}
        plugins={[Fullscreen, Captions, Share]}
      />

      // ...
    </>
  );
}
```

p.s. I've added a note to simplify this process in one of the future releases.

**Reply by Flo2410** [reply]
Thank you very much for the fast answer.

**Reply by igordanchenko** [reply]
With v3.3.0 you can now override plugin's button position:

```jsx
<Lightbox
  toolbar={{
    buttons: [
      <button type="button" className="yarl__button" key={"share_button"}>
        <BiShareAlt className="p-1 yarl__icon" />
      </button>,
      "fullscreen",
      "close",
    ],
  }}
  plugins={[Fullscreen, Captions]}
  // ...
/>
```

**Reply by Flo2410** [reply]
Nice, thank you!
